libs/books/feature/src/lib/reading-list/reading-list.component.ts

```diff
@@ -9,10 +11,53 @@ import { getReadingList, removeFromReadingList } from '@tmo/books/data-access';
 })
 export class ReadingListComponent {
   readingList$ = this.store.select(getReadingList);
+  book:Book;
+  readingItem:ReadingListItem;
+  readingItems:ReadingListItem[];
+  constructor(private readonly store: Store, private snackBar: MatSnackBar) {}
 
-  constructor(private readonly store: Store) {}
-
-  removeFromReadingList(item) {
+  removeFromReadingList(item : ReadingListItem) {
     this.store.dispatch(removeFromReadingList({ item }));
+    
+    this.book={
+      id:'',
+      title:'', 
+      authors:[],
+      description:'',
+      coverUrl:'',
+      publisher:''
+    }
+    this.book.id=item.bookId;
+    this.book.authors=item.authors;
+    this.book.title=item.title;
+    this.book.description=item.description;
+    this.book.coverUrl=item.coverUrl;
+    this.book.publisher=item.publisher;
+    
+    this.openSnackBar('Book Removed','Undo');
+  }
+  openSnackBar(message: string, action: string) {
+    const snackbar=this.snackBar.open(message, action,{
+      duration: 3000
+    });
+    
+    snackbar.onAction().subscribe(() => {
+      this.store.dispatch(addToReadingList({book:this.book} ));
+    });
+   
+  }
+  finishedBook(item:ReadingListItem)
+  {
+    this.readingItem={
```

Comment:
Same here, just dispatch the new item without store the transient state on the class instance.